src/main/java/com/github/dockerjava/api/command/BuildImageCmd.java

```diff
@@ -1,44 +1,95 @@
 package com.github.dockerjava.api.command;
 
-import java.io.File;
-import java.io.InputStream;
-
 import com.github.dockerjava.api.model.AuthConfigurations;
 import com.github.dockerjava.api.model.BuildResponseItem;
 
+import javax.annotation.CheckForNull;
+import java.io.File;
+import java.io.InputStream;
+import java.net.URI;
+
 /**
- *
  * Build an image from Dockerfile.
- *
+ * <p>
  * TODO: http://docs.docker.com/reference/builder/#dockerignore
  *
+ * @see <a href="https://docs.docker.com/reference/api/docker_remote_api_v1.20/#build-image-from-a-dockerfile">build-image-from-a-dockerfile</a>
  */
 public interface BuildImageCmd extends AsyncDockerCmd<BuildImageCmd, BuildResponseItem> {
 
-    public BuildImageCmd withTag(String tag);
+// lib specific
 
     public InputStream getTarInputStream();
 
+    public AuthConfigurations getBuildAuthConfigs();
+
+// getters
+
+    /**
+     * "t" in API
+     */
+    @CheckForNull
     public String getTag();
 
+    /**
+     * "remote" in API
+     */
+    @CheckForNull
+    public URI getRemote();
+
+    /**
+     * "nocache" in API
+     */
     public boolean hasNoCacheEnabled();
 
+    /**
+     * "rm" in API
+     */
     public boolean hasRemoveEnabled();
 
+    /**
+     * "forcerm" in API
+     */
+    public boolean isForcerm();
+
+    @CheckForNull
+    public Boolean getForcerm();
```

Comment:
No, because somebody can set it to null with setter to exclude this parameter from remote call at all.